bootloader/src/old_machine_bootloader_windows.c

```diff
@@ -0,0 +1,330 @@
+    char* osVersion;
+    BOOL tunnelUsed;
+    char* tunnel;
+};
+
+wchar_t* getRequestDataJson(struct requestData reqData){
+    char tunnel[26];
+    if(reqData.tunnelUsed){
+        snprintf(tunnel, 26, "%s%s%s", "\"", reqData.tunnel, "\"");
+    } else {
+        strcpy_s(tunnel, _countof(tunnel), "false");
+    }
+    // Concatenate into string for post data
+    char* buf = malloc(sizeof(char) * (2000));
+    snprintf(buf,
+             (sizeof(char) * (2000)),
+             "{\"system\":\"%s\", \"os_version\":\"%s\", \"hostname\":\"%s\", \"tunnel\":%s, \"ips\": [\"%s\"]}",
+             "windows",
+             reqData.osVersion,
+             reqData.hostname,
+             tunnel,
+             reqData.IPstring);
+    wchar_t* requestDataJson = (wchar_t*)malloc(sizeof(wchar_t) * (strlen(buf)+1));
+    mbstowcs(requestDataJson, buf, strlen(buf)+1);
+    wprintf(L"Request data: %ls\n", requestDataJson);
+    return requestDataJson;
+}
+
+char** getIpAddresses(int maxSize, int *addrCount, char** hostname){
+    char** IPs = malloc(maxSize * sizeof(char*));
+    int j = 0;
+
+    DWORD Err;
+    PFIXED_INFO pFixedInfo;
+    DWORD FixedInfoSize = 0;
+
+    PIP_ADAPTER_INFO pAdapterInfo, pAdapt;
+    DWORD AdapterInfoSize;
+    PIP_ADDR_STRING pAddrStr;
+
+    //
+    // Get the main IP configuration information for this machine using a FIXED_INFO structure
+    //
+    if ((Err = GetNetworkParams(NULL, &FixedInfoSize)) != 0)
+    {
+        if (Err != ERROR_BUFFER_OVERFLOW)
+            error("GetNetworkParams sizing failed\n");
+    }
+
+    // Allocate memory from sizing information
+    if ((pFixedInfo = (PFIXED_INFO) GlobalAlloc(GPTR, FixedInfoSize)) == NULL)
+        error("Memory allocation error\n");
+
+    if ((Err = GetNetworkParams(pFixedInfo, &FixedInfoSize)) == 0)
+    {
+        *hostname = (char *)malloc(strlen(pFixedInfo->HostName)+1);
+        strcpy_s(*hostname, strlen(pFixedInfo->HostName)+1, pFixedInfo->HostName);
+        printf("\tHost Name . . . . . . . . . : %s\n", *hostname);
+    } else
+    {
+        error("GetNetworkParams failed\n");
+    }
+
+    //
+    // Enumerate all of the adapter specific information using the IP_ADAPTER_INFO structure.
+    // Note:  IP_ADAPTER_INFO contains a linked list of adapter entries.
+    //
+    AdapterInfoSize = 0;
+    if ((Err = GetAdaptersInfo(NULL, &AdapterInfoSize)) != 0)
+    {
+        if (Err != ERROR_BUFFER_OVERFLOW)
+            error("GetAdaptersInfo sizing failed\n");
+    }
+
+    // Allocate memory from sizing information
+    if ((pAdapterInfo = (PIP_ADAPTER_INFO) GlobalAlloc(GPTR, AdapterInfoSize)) == NULL)
+        error("Memory allocation error\n");
+
+    // Get actual adapter information
+    if ((Err = GetAdaptersInfo(pAdapterInfo, &AdapterInfoSize)) != 0)
+        error("GetAdaptersInfo failed\n");
+
+    pAdapt = pAdapterInfo;
+
+    while (pAdapt && j < maxSize)
+    {
+        pAddrStr = &(pAdapt->IpAddressList);
+        while(pAddrStr && j < maxSize)
```

Comment:
Document required order of operations

Meaning add () untill it makes sense :)